ftw/contacts/browser/contact.pt

```diff
@@ -0,0 +1,113 @@
+                <th i18n:translate="label_function">Function</th>
+                <td tal:content="context/function">Boss</td>
+              </tr>
+              <tr tal:condition="context/organization">
+                <th i18n:translate="label_organization">Organization</th>
+                <td tal:content="context/organization">XYZ GmbH</td>
+              </tr>
+              <tr tal:condition="context/department">
+                <th i18n:translate="label_department">Department</th>
+                <td tal:content="context/department">CIA</td>
+              </tr>
+              <tr tal:condition="python: context.postal_code and context.city and context.address">
+                <th i18n:translate="label_address">Address</th>
+                <td>
+                  <span tal:replace="structure python: context.address and context.address.replace(u'\n', u'<br />') or ''" /><br />
+                  <span tal:replace="context/postal_code" />
+                  <span tal:replace="context/city" />
+                </td>
+              </tr>
+              <tr tal:condition="context/phone_office">
+                <th i18n:translate="label_phone_office">Office phone number</th>
+                <td tal:content="context/phone_office">031 000 00 00</td>
+              </tr>
+              <tr tal:condition="context/phone_mobile">
+                <th i18n:translate="label_phone_mobile">Mobile phone number</th>
+                <td tal:content="context/phone_mobile">079 000 00 00</td>
+              </tr>
+              <tr tal:condition="context/fax">
+                <th i18n:translate="label_fax">Fax number</th>
+                <td tal:content="context/fax">031 123 45 67</td>
+              </tr>
+              <tr tal:condition="context/email">
+                <th i18n:translate="label_email">E-Mail</th>
+                <td>
+                  <a tal:attributes="href string:mailto:${context/email}"
+                  tal:content="context/email">info@mail.com</a>
+                </td>
+              </tr>
+              <tr tal:condition="context/www">
+                <th i18n:translate="label_www">www</th>
+                <td>
+                <a tal:condition="context/www"
+                   tal:content="python: context.www.replace('https://', '').replace('https://', '')"
+                   tal:attributes="href context/www"
+                   target="_blank">www.google.ch</a>
+                </td>
+              </tr>
+            </table>
+            <tal:private condition="context/address_private">
+              <h2 i18n:translate="label_private_address">Privateaddress</h2>
+              <table class="contactPrivateInfo">
+                <tr>
+                  <th i18n:translate="label_address">Address</th>
+                  <td>
+                    <span tal:replace="structure python: context.address_private and context.address_private.replace(u'\n', u'<br />') or ''" /><br />
```

Comment:
I'd prefer that a text -> safe-html transform is used in such situations.

Using "structure" creates a XSS leak: I could easily store a `<script>` tag in the private address field, which would then not be escaped and executed in the browser of any visitor.
By using the regular transform HTML tags would be escaped.